pyscf/neo/grad.py

```diff
@@ -9,6 +9,137 @@
+    '''evaluate e_xc and v_xc of proton on a grid (epc17)'''
+    a = 2.35
+    b = 2.4
+
+    if epc == '17-1':
+        c = 3.2
+    elif epc == '17-2':
+        c = 6.6
+    elif isinstance(epc, dict):
+        a = epc["a"]
+        b = epc["b"]
+        c = epc["c"]
+    else:
+        raise ValueError('Unsupported type of epc %s', epc)
+
+    rho_product = numpy.multiply(rho_e, rho_n)
+    denominator = a - b * numpy.sqrt(rho_product) + c * rho_product
+    denominator = numpy.square(denominator)
+    numerator = -a * rho_e + numpy.multiply(numpy.sqrt(rho_product), rho_e) * b * 0.5
+    vxc = numpy.multiply(numerator, 1 / denominator)
+
+    return vxc
+
+def eval_xc_elec(epc, rho_e, rho_n):
+    '''evaluate e_xc and v_xc of electrons on a grid (only the epc part)'''
+    a = 2.35
+    b = 2.4
+
+    if epc == '17-1':
+        c = 3.2
+    elif epc == '17-2':
+        c = 6.6
+    elif isinstance(epc, dict):
+        a = epc["a"]
+        b = epc["b"]
+        c = epc["c"]
+    else:
+        raise ValueError('Unsupported type of epc %s', epc)
+
+    rho_product = numpy.multiply(rho_e, rho_n)
+    denominator = a - b * numpy.sqrt(rho_product) + c * rho_product
+    denominator = numpy.square(denominator)
+    numerator = -a * rho_n + numpy.multiply(numpy.sqrt(rho_product), rho_n) * b * 0.5
+    vxc = numpy.multiply(numerator, 1 / denominator)
+
+    return vxc
+
+
+def get_vepc_elec(mf_grad):
+    mf = mf_grad.base
+    mol = mf_grad.mol
+    ni = mf.mf_elec._numint
+    grids = mf.mf_elec.grids
+    nao = mol.elec.nao_nr()
+    ao_loc = mol.elec.ao_loc_nr()
+    ao_deriv = 1
+    vmat_elec = numpy.zeros((3,nao,nao))
+    for i in range(mol.nuc_num):
+        ia = mol.nuc[i].atom_index
+        if mol.atom_pure_symbol(ia) == 'H' and \
+            (isinstance(mf.epc, str) or ia in mf.epc['epc_nuc']):
+            for _, mask, weight, coords \
+                in ni.block_loop(mol.elec, grids, nao, ao_deriv):
+                ao_nuc = eval_ao(mol.nuc[i], coords, ao_deriv)
+                rho_nuc = eval_rho(mol.nuc[i], ao_nuc[0], mf.dm_nuc[i])
+                rho_nuc[rho_nuc<0.] = 0.
+                ao_elec = eval_ao(mol.elec, coords, ao_deriv)
+                rho_elec = eval_rho(mol.elec, ao_elec[0], mf.dm_elec)
+                vxc_elec_i = eval_xc_elec(mf.epc, rho_elec, rho_nuc)
+                aow_elec = _scale_ao(ao_elec[0], weight * vxc_elec_i)
+                _d1_dot_(vmat_elec, mol.elec, ao_elec[1:4], aow_elec, mask, ao_loc, True)
+    return -vmat_elec
+
+def get_vepc_nuc(mf_grad, mol, dm):
+    mf = mf_grad.base
+    ni = mf.mf_elec._numint
+    grids = mf.mf_elec.grids
+    nao = mol.nao_nr()
+    ao_loc = mol.ao_loc_nr()
+    # add electron epc grad
+    ao_deriv = 1
+    vmat_nuc = numpy.zeros((3,nao,nao))
+    for _, mask, weight, coords \
+        in ni.block_loop(mol, grids, nao, ao_deriv):
+        ao_nuc = eval_ao(mol, coords, ao_deriv)
+        rho_nuc = eval_rho(mol, ao_nuc[0], dm)
+        rho_nuc[rho_nuc<0.] = 0.
```

Comment:
Are you sure this is correct? You want to make rho_nuc positive, but do not make its gradients positive.